### PR TITLE
HOLO-1365 / Explore Viem For Holograph SDK

### PR DESCRIPTION
**Pull request by BravoNatalie**
## Describe Changes

Changes made to the package: **sdk**.

I made it better by doing ...

- Remove `ethers`
- Add `viem`
- Add tests for decoders functions
- Add tests for encoders functions
- Add `getLogs` script

image

image
image

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All tests are passing

### CODE REVIEW

File: packages/sdk/scripts/example.ts [hunk 1]
```diff
@@ -23,15 +23,25 @@ async function main() {
...
 
   const holograph = new Holograph(config)
   console.log(await holograph.getRegistryByNetworks())
 
   /// Multi-providers usage:
 
-  const multiProviders = new Providers(config)
-  console.log('providers: ', multiProviders.providers)
+  // const multiProviders = new Providers(config)
+  // console.log('providers: ', multiProviders.providers)
```
Comment (feliperodriguess) [inline packages/sdk/scripts/example.ts:44]: Can we remove these comments?

File: packages/sdk/src/services/holograph-protocol.service.ts [hunk 2]
```diff
@@ -9,20 +9,17 @@ export class HolographProtocol {
   private holographContract!: Holograph
   private registryContract!: Registry
   private treasuryContract!: Treasury
-  private operatorContract!: Operator
```
Comment (feliperodriguess) [inline packages/sdk/src/services/holograph-protocol.service.ts:12]: Why did you remove all the `Operator` references?

**Review by feliperodriguess — CHANGES_REQUESTED** [review]

File: packages/sdk/src/contracts/holograph.contract.ts [hunk 3]
```diff
@@ -60,18 +59,19 @@ export class Holograph {
     const address = this.getAddress(chainId)
 
     logger.info({chainId, address}, 'getting holograph contract')
-    const contract = getContract({address, abi: HolographABI, signerOrProvider: provider})
+    const contract = getContract({address, abi: HolographABI, client: provider})
 
     let result
     try {
-      result = await contract[functionName](...args)
+      // @ts-ignore
```
[the remark this answers is not in the document]
Comment (BravoNatalie) [reply packages/sdk/src/contracts/holograph.contract.ts]: Couldn't get it to work, it's complaining about conflicting types.

**Review by feliperodriguess — APPROVED** [review]

**Review by sogoiii — APPROVED** [review]
👍 @BravoNatalie i love the encoder and decoder tests

**Pull request merged**